code/parse/sexp.cpp

```diff
@@ -17367,6 +17369,59 @@ int sexp_goal_incomplete(int n)
 		return SEXP_TRUE;
 }
 
+/**
+ * Resets an event, its status, and its nodes so that it is as if the event had never been evaluated.
+ */
+void sexp_reset_event(int node)
+{
+	for (int n = node; n >= 0; n = CDR(n))
+	{
+		auto name = CTEXT(n);
+		int event_num = mission_event_lookup(name);
+		if (event_num >= 0)
+		{
+			auto eventp = &Mission_events[event_num];
+
+			eventp->result = 0;
+			eventp->timestamp = TIMESTAMP::invalid();
+
+			// clear some flags but not all
+			eventp->flags &= ~MEF_CURRENT;
+			eventp->flags &= ~MEF_DIRECTIVE_SPECIAL;
+			eventp->flags &= ~MEF_DIRECTIVE_TEMP_TRUE;
+			eventp->flags &= ~MEF_TIMESTAMP_HAS_INTERVAL;
+
+			eventp->count = 0;
+			eventp->satisfied_time = TIMESTAMP::invalid();
+			eventp->born_on_date = TIMESTAMP::invalid();
+			eventp->previous_result = 0;
+
+			if (eventp->formula >= 0)
```

Comment:
`flush_sexp_tree` already short-circuits on a negative node index.

(More curiosity than anything, but when would an event not have a formula?)

Reply:
So it does.  Tweaked.

Hmm... it looks like an event's formula is set to -1 when it is failed, or known-false.  That will need to be modified for this PR to work.